SQL/when-to-add-indexes.md

```diff
@@ -0,0 +1,66 @@
+# When to add indexes
+author: SebaRaba
+
+levels:
+
+  - beginner
+
+  - basic
+
+type: normal
+
+category: must-know
+
+links:
+
+  - '[More on CRETE INDEX](https://www.tutorialspoint.com/sql/sql-indexes.htm)'
+
+---
+## Content
+
+Indexes are meant to help SQL Servers to lookup data *faster*. However we need to pay attention at how often the data will be inserted, updated and deleted. That is because each time we modify data in an indexed table, the index needs to be updated as well. This can lead to *lower performance* in the end.
+
+Thus, we should bear in mind the followings while creating indexes:
+- for tables that are heavily updated, use as few columns as possible under the same index
+- if the table contains a lot of data, but the data isn't modified that frequently, use as many indexes to improve performance
+- try to keep columns as short as possible in clustered tables
```

Comment:
Do you mean, pick columns with very little data (numbers, short strings), as opposed to something like `text`? Or do you mean, don't use very many columns?